src/h3/mod.rs

```diff
@@ -1400,7 +1402,7 @@ impl Connection {
 
 /// Generates an HTTP/3 GREASE variable length integer.
 fn grease_value() -> u64 {
-    let n = std::cmp::min(super::rand::rand_u64(), 148_764_065_110_560_899);
+    let n = rand::thread_rng().gen_range(0, 148_764_065_110_560_900);
```

Comment:
Based on @ghedo 's comment on #146 , I think this is most critical change I guess.
Previous code try to make a random range [0, u64_max] but have a max bound of 148_764_065_110_560_899, which only 0.8% of u64_max, make it highly skewed to 148_764_065_110_560_900 (more than 99% will have this value, roughly matching to @LPardue's test result)

If so, is `super::rand::rand_u64() % 148_764_065_110_560_900` good enough, without random API changes?

Reply:
I opted for the suggestion in Ghedo's comment "Integer only w/ re-rolls" method cf.  https://codereview.stackexchange.com/a/159760.

This apparently addresses shortcoming using modulo. The result of this method is 

```
0.00020000020000019998'th percentile of data is 2199023255551 with 2 samples
0.0003000003000003'th percentile of data is 8796093022207 with 1 samples
0.0007000007000007'th percentile of data is 17592186044415 with 4 samples
0.0009000009000009001'th percentile of data is 35184372088831 with 2 samples
0.0014000014000014'th percentile of data is 70368744177663 with 5 samples
0.0032000032000031996'th percentile of data is 140737488355327 with 18 samples
0.0062000062000062'th percentile of data is 281474976710655 with 30 samples
0.0132000132000132'th percentile of data is 562949953421311 with 70 samples
0.0262000262000262'th percentile of data is 1125899906842623 with 130 samples
0.05000005000005'th percentile of data is 2251799813685247 with 238 samples
0.1002001002001002'th percentile of data is 4503599627370495 with 502 samples
0.1952001952001952'th percentile of data is 9007199254740991 with 950 samples
0.3863003863003863'th percentile of data is 18014398509481983 with 1911 samples
0.7789007789007789'th percentile of data is 36028797018963967 with 3926 samples
1.5625015625015626'th percentile of data is 72057594037927935 with 7836 samples
3.130903130903131'th percentile of data is 144115188075855871 with 15684 samples
6.245906245906245'th percentile of data is 288230376151711743 with 31150 samples
12.48901248901249'th percentile of data is 576460752303423487 with 62431 samples
25.00782500782501'th percentile of data is 1152921504606846975 with 125188 samples
49.99844999845'th percentile of data is 2305843009213693951 with 249906 samples
100'th percentile of data is 4611686018427387903 with 500015 samples
```

Reply:
@junhochoi the modulo operation is biased, so results are not uniform. Though I guess in this case it wouldn't matter too much, the proper solution is not that much more complicated.